### PR TITLE
document the CI pipeline

### PR DESCRIPTION
**Pull request by monperrus**

### CODE REVIEW

**Comment by MartinWitt** [comment]
I will add a paragraph about the CD we have.

**Comment by MartinWitt** [comment]
I have added a section about the CD process and tried to document the important maven central releases in more detail.

File: doc/ci-cd.md [hunk 1]
```diff
@@ -1,4 +1,58 @@
...
+  identical using [diffoscope](https://diffoscope.org/)
+
+More commands exist for testing code coverage, ensuring the various spoon
+submodules still compile and their tests pass, as well as helpers for releasing
+spoon versions.
+
+## Continuous delivery
+
+Spoon has three different release channels:
+- **Snapshot**: The latest development version, published to the Sonatype
```
Comment (I-Al-Istannen) [inline doc/ci-cd.md:32]:
```suggestion
- **Snapshot**: The latest development version, published daily to the Sonatype
```

File: doc/ci-cd.md [hunk 2]
```diff
@@ -1,4 +1,58 @@
...
+  snapshot repository. This version is not guaranteed to be stable, and may
```
Comment (I-Al-Istannen) [inline doc/ci-cd.md:33]:
```suggestion
  snapshot repository. This version is not guaranteed to be stable and may
```

File: doc/ci-cd.md [hunk 3]
```diff
@@ -1,4 +1,58 @@
...
+  contain breaking changes. After 90 days, snapshots are automatically deleted by Sonatype.
+- **Beta**: A release candidate for the next stable release. Weekly beta
+  releases are published to the SonaType release repository. This may contain unstable features, but should be mostly stable.
```
Comment (I-Al-Istannen) [inline doc/ci-cd.md:36]:
```suggestion
  releases are published to the SonaType release repository. They may contain unstable features, but should be mostly stable.
```

File: doc/ci-cd.md [hunk 4]
```diff
@@ -1,4 +1,58 @@
...
+- **Stable**: A stable release following the [Semantic Versioning](https://semver.org)
+  specification. These releases are published to the Sonatype release
+  repository.
+
+For the automatic release process, we use the [JReleaser](https://jreleaser.org) and GitHub actions.
+Beta and Snapshot releases are automatically published, see [Beta](/.github/workflows/release-beta.yml) and [Snapshot](/.github/workflows/release-nightly.yml) workflows.
```
Comment (I-Al-Istannen) [inline doc/ci-cd.md:42]:
```suggestion
Beta and Snapshot releases are published automatically, see [Beta](/.github/workflows/release-beta.yml) and [Snapshot](/.github/workflows/release-nightly.yml).
```

File: doc/ci-cd.md [hunk 5]
```diff
@@ -1,4 +1,58 @@
...
+Stable releases are manually triggered, see [SemVer](/.github/workflows/release-manual.yml) workflow.
+This is a GitHub workflow that triggers the [JReleaser](https://jreleaser.org) release process.
```
Comment (I-Al-Istannen) [inline doc/ci-cd.md:44]:
```suggestion
Stable releases are triggered manually using the [SemVer](/.github/workflows/release-manual.yml) workflow, which starts the [JReleaser](https://jreleaser.org) release process.
```

**Comment by MartinWitt** [comment]
@monperrus feel free to merge when you are happy with the text.

**Comment by monperrus** [comment]
thanks a lot!

**Pull request merged**